app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventScreen.kt

```diff
@@ -139,8 +142,15 @@ fun AddEventScreenContent(
     val singlePhotoPickerLauncher =
         rememberLauncherForActivityResult(
             contract = ActivityResultContracts.PickVisualMedia(),
-            onResult = { uri ->
-                uri?.let {
+            onResult = { uriNullable ->
+
+                uriNullable?.let { uri ->
+
+                    if(uri.isImageSizeTooLargeToUpload(context, UPLOAD_IMAGE_MAX_SIZE)) {
```

Comment:
Here, I'd actually not make this an extension function of `Uri`, since I'd then expect this to be generic behavior that applies to all Uris, but it only applies to event photo Uris. This logic is rather specific to AgendaItems/Events, so maybe a function in a companion object in one of these classes could make more sense?
```
Event.isImageSizeTooLargeToUpload(uri, context, UPLOAD_IMAGE_MAX_SIZE)
```

Or you create something like a `UriByteConverter`:
```
// In domain
interface UriByteConverter {
    fun convertToBytes(uri: Uri, maxSize: Long): ByteArray
}

// In data
class UriByteConverter(private val context: Context): UriByteConverter {
    override fun convertToBytes(uri: Uri, maxSize: Long): ByteArray {
        ...
    }
}
```

Reply:
OK will refactor this for correct package